packages/ui-plugin-sheets/src/views/status-bar/StatusBar.tsx

```diff
@@ -0,0 +1,142 @@
+import { useDependency } from '@wendellhu/redi/react-bindings';
+import clsx from 'clsx';
+import { useEffect, useState } from 'react';
+
+import { IStatusBarService } from '../../services/status-bar.service';
+import { CopyableStatisticItem, formatNumber, functionDisplayNames, IStatisticItem } from './CopyableStatisticItem';
+import styles from './index.module.less';
+
+const SINGLE_MODE_WIDTH = 800;
+
+export const StatusBar = () => {
+    const [isSingle, setIsSingle] = useState(window.innerWidth < SINGLE_MODE_WIDTH);
+    const [show, setShow] = useState(true);
+
+    const localeService = useDependency(LocaleService);
+    const statusBarService = useDependency(IStatusBarService);
+    const items = statusBarService.getFunctions().map((item, index) => ({
+        name: item,
+        value: 0,
+        show: isSingle ? index < 1 : index < 3,
+    }));
+    const [statistics, setStatistics] = useState<IStatisticItem[]>(items);
+    const showList = statistics.filter((item) => item.show);
+
+    const onChange = (item: IStatisticItem) => {
+        const newStatistics = statistics.map((stat) => {
+            if (stat.name === item.name) {
+                stat.show = !stat.show;
+            } else if (isSingle) {
+                stat.show = false;
+            }
+            return stat;
+        });
+        setStatistics(newStatistics);
+    };
+
+    useEffect(() => {
+        const subscription = statusBarService.state$.subscribe((item) => {
+            if (!item) {
+                setShow(false);
+            } else {
+                setShow(true);
+                const newStatistics = statistics.map((stat) => {
+                    const target = item.find((i) => i.func === stat.name);
+                    stat.value = target?.value ?? stat.value;
+                    return stat;
+                });
+                setStatistics(newStatistics);
+            }
+        });
+        return () => {
+            subscription.unsubscribe();
+        };
+    }, [statusBarService, statistics]);
+
+    useEffect(() => {
+        const handleResize = () => {
+            const newSingleState = window.innerWidth < SINGLE_MODE_WIDTH;
+            if (isSingle !== newSingleState) {
+                if (newSingleState) {
+                    // keep the items hide except the first show item
+                    const firstIndex = statistics.findIndex((item) => item.show);
+                    const newStatistics = statistics.map((stat, index) => {
+                        if (index === firstIndex) {
+                            stat.show = true;
+                        } else {
+                            stat.show = false;
+                        }
+                        return stat;
+                    });
+                    setStatistics(newStatistics);
+                }
+                setIsSingle(newSingleState);
+            }
+        };
+
+        window.addEventListener('resize', handleResize);
```

Comment:
@jikkai The UI plugin should expose a service for components to subscribe to layout changes (not only window resize event).